### PR TITLE
Add quickstart guide for Javascript consumers

### PR DESCRIPTION
**Pull request by TimothyJones**
This PR includes a quickstart guide for Javascript Consumers using Jest (which is everyone using React). Would appreciate a few eyeballs over it to check I haven't included anything unnecessarily complex (@mefellows ?)

### CODE REVIEW

File: implementation_guides/javascript.md [hunk 1]
```diff
@@ -2,13 +2,119 @@
...
+      pactfileWriteMode: 'update',
+      consumer: /* the name of your consumer */,
+      provider: /* the name of your provider */
+    });
+
+ Global setup for Jest is handled with `pactTestWrapper.js`:
+
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000; //* This is to give the pact mock server time to start
+
+    beforeAll(done => {
```
Comment (mefellows) [inline implementation_guides/javascript.md]: Can this not just be `beforeAll(() => provider.setup())`?

File: implementation_guides/javascript.md [hunk 2]
```diff
@@ -2,13 +2,119 @@
...
+          return provider.addInteraction(interaction);
+        });
+
+        // add expectations
+        it(/* describe the test */, done => {
+          api(url).someCall()
+            .then(response => {
+              expect(response).toEqual(/* check the response here, using the default values provided to the Pact DSL */);
+            })
+            .then(() => provider.verify())
```
Comment (mefellows) [inline implementation_guides/javascript.md]: It's probably OK here, but as discussed in Slack, I like separating this from the test itself

Comment (mefellows) [reply implementation_guides/javascript.md]: I'm pretty sure `afterEach(() => provider.verify())`  should work.

File: implementation_guides/javascript.md (hunk 1 above)
Comment (TimothyJones) [reply implementation_guides/javascript.md]: It can! Changed.

File: implementation_guides/javascript.md (hunk 2 above)
Comment (TimothyJones) [reply implementation_guides/javascript.md]: Good catch.

**Pull request merged**